lib/unimatrix/authorization/filters/requires_policies.rb

```diff
@@ -70,29 +68,34 @@ def self.included( controller )
       controller.extend( ClassMethods )
     end
 
-    def policies=(attributes)
+    def policies=( attributes )
       @policies = attributes
     end
 
     def policies
       @policies ||= begin
-        retrieve_policies( controller_name, params[ :access_token ], realm_uuid )
+        # Used by Archivist requires_permission filter. Todo: deprecate
+        retrieve_policies( @resource_name, params[ :access_token ], realm_uuid )
       end
     end
 
-    def retrieve_policies( resource_name, access_token, realm )
-      if resource_name && access_token && realm
-        resource  = "realm/#{ realm }::#{ ENV['APPLICATION_NAME'] }::#{ resource_name }/*"
-        params    = "resource=#{ resource }&access_token=#{ access_token }"
-        request_policies( params )
+    # In Rails app, this is overwritten by #retrieve_policies in railtie.rb
+    def retrieve_policies( resource_name, access_token, realm_uuid )
+      if resource_name && access_token
+        request_policies( resource_name, access_token, realm_uuid )
       end
     end
 
-    def request_policies( params )
-      uri = URI.parse( "#{ ENV['KEYMAKER_URL'] }/policies?#{ params }" )
-      response = Net::HTTP.get( uri )
-      JSON.parse( response )[ 'policies' ] rescue nil
+    def request_policies( resource_name, access_token, realm_uuid )
+      if resource_name && access_token
+        realm_uuid = realm_uuid || '*'
```

Comment:
The issue with writing realms I brought up here hasn't been addressed https://github.com/bedrocketjmd/unimatrix-keymaker-sdk/pull/7#discussion_r88712459

The query for ```resource: "realm/*::regent::realms/*"``` will return all policies for updating realms. So if i user has a policy to update realm 123, but not the permission to create/update all realms (*), this will still return the policy for realm 123 and they will be allowed to create a realm even though they shouldn't.